payara-management-api/src/main/java/fish/payara/arquillian/management/ExternalServerManagement.java

```diff
@@ -0,0 +1,87 @@
+ *    https://github.com/payara/Payara/blob/master/LICENSE.txt
+ *    See the License for the specific
+ *    language governing permissions and limitations under the License.
+ *
+ *    When distributing the software, include this License Header Notice in each
+ *    file and include the License file at glassfish/legal/LICENSE.txt.
+ *
+ *    GPL Classpath Exception:
+ *    The Payara Foundation designates this particular file as subject to the "Classpath"
+ *    exception as provided by the Payara Foundation in the GPL Version 2 section of the License
+ *    file that accompanied this code.
+ *
+ *    Modifications:
+ *    If applicable, add the following below the License Header, with the fields
+ *    enclosed by brackets [] replaced by your own identifying information:
+ *    "Portions Copyright [year] [name of copyright owner]"
+ *
+ *    Contributor(s):
+ *    If you wish your version of this file to be governed by only the CDDL or
+ *    only the GPL Version 2, indicate your decision by adding "[Contributor]
+ *    elects to include this software in this distribution under the [CDDL or GPL
+ *    Version 2] license."  If you don't indicate a single choice of license, a
+ *    recipient has the option to distribute your version of this file under
+ *    either the CDDL, the GPL Version 2 or to extend the choice of license to
+ *    its licensees as provided above.  However, if you add GPL Version 2 code
+ *    and therefore, elected the GPL Version 2 license, then the option applies
+ *    only if the new code is made subject to such option by the copyright
+ *    holder.
+ */
+
+package fish.payara.arquillian.management;
+
+import fish.payara.arquillian.management.impl.PayaraManagementExtension;
+import fish.payara.arquillian.management.impl.StaticPayaraManagementLookup;
+
+import java.util.List;
+
+/**
+ * Management operations to be performed from client side, before test run.
+ *
+ */
+public interface ExternalServerManagement {
+    String getAdminHost();
+    int getAdminPort();
+    String getDomainName();
+    String getPayaraVersion();
+
+    boolean isRunning();
+
+    void restartDomain();
+
+    void restartInstance(String instanceName);
+
+    List<String> listInstances();
+
+    /**
+     * A container MAY support running arbitrary commands, e.g. my means of invoking asadmin script or remote REST
+     * interface, the set of supported commands might be limited, or non-existing. For example, Payara micro might
+     * not support those.
+     * @param commandName
+     * @param parameters
+     * @return result of the command
+     * @throws UnsupportedOperationException when container cannot run arbitrary commands outside the server.
+     */
+    CommandResult asadmin(String commandName, String... parameters);
+
+    /**
+     * Register a command to be executed after server startup.
+     * @param commandName
+     * @param parameters
+     */
+    void postBootCommand(String commandName, String... parameters);
```

Comment:
Other command methods like `preBootCommand` & `postDeployCommand` can also be introduced in this API.

Reply:
Yes, I'd actually start with even smaller set of methods than that first and implement it consistently across containers. Then we'd have enough experience to get feeling on what the API should look like.